module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -129,12 +129,130 @@ public function addCss($css)
      */
     public function addJs($js)
     {
-        if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+        if ((!is_array($js) && !is_a($js, 'Traversable')) || isset($js['file'])) {
+            $this->addJsEntry($js);
+        } elseif (isset($js[0])) {
+            foreach ($js as $current) {
+                $this->addJsEntry($current);
+            }
+        } elseif ($js == []) {
+            return;
+        } else {
+            throw new \Exception("Invalid JS entry format: " . print_r($js, true));
+        }
+    }
+
+    /**
+     * Helper function for adding a Javascript file.
+     *
+     * @param string|array $jsEntry Entry to add, either as string with path
+     * or array with additional properties.
+     *
+     * @return void
+     */
+    protected function addJsEntry($jsEntry)
+    {
+        if (!is_array($jsEntry)) {
+            $this->addJsStringEntry($jsEntry);
+        } else {
+            $this->addJsArrayEntry($jsEntry);
         }
-        foreach ($js as $current) {
-            $this->js[] = $current;
+    }
+
+    /**
+     * Helper function for adding a Javascript file which is described as string.
+     *
+     * @param string $jsEntry Entry to add as string.
+     *
+     * @return void
+     */
+    protected function addJsStringEntry($jsEntry)
+    {
+        $parts = $this->parseSetting($jsEntry);
+        if (count($parts) == 1) {
+            $jsEntry = ['file' => $jsEntry];
+        } else {
+            $jsEntry = [
+                'file' => $parts[0],
+                'attributes' => ['conditional' => trim($parts[1])],
+            ];
         }
+        $this->addJsArrayEntry($jsEntry);
+    }
+
+    /**
+     * Helper function for adding a Javascript file which is described as array.
+     *
+     * @param string $jsEntry Entry to add as string.
+     *
+     * @return void
+     */
+    protected function addJsArrayEntry($jsEntry)
+    {
+        if (isset($jsEntry['priority']) && isset($jsEntry['load_after'])) {
+            throw new \Exception(
+                'Using "priority" as well as "load_after" in the same entry '
+                . 'is not supported: "' . $jsEntry['file'] . '"'
+            );
+        }
+
+        foreach ($this->js as $existingEntry) {
+            if ($existingEntry['file'] == $jsEntry['file']) {
+
+                // If we have the same settings as before, just skip this entry.
+                if (count(array_diff($existingEntry, $jsEntry)) > 0) {
```

Comment:
Is this doing the right thing? I may be missing something, but for the settings to be same array_diff would return an empty array, no? Can we be sure both entries have all the same keys? Why `count` when `!empty` would do?

Reply:
You're right! array_diff seems to be a problem if the values are the same but the keys aren't!

After some more tests, i propose we simply use `==` instead to compare the arrays. I would not recommend using `===` because that would only return true if the keys are in the same order.